src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -95,6 +98,21 @@ def refresh_repo(self):
                 
             raise
 
+    def __refresh_repo_services(self):
+        """ Similar to refresh_repo, but refreshes services in case no repos are defined. """
+        self.composite_logger.log("Refreshing local repo services...")
+        try:
+            self.invoke_package_manager(self.repo_refresh_services)
+        except Exception as error:
+            # Reboot if not already done
+            if self.status_handler.get_installation_reboot_status() == Constants.RebootStatus.COMPLETED:
+                self.composite_logger.log_warning("Unable to refresh repo services (retries exhausted after reboot).")
+            else:
+                self.composite_logger.log_warning("Setting force_reboot flag to True after refreshing repo services.")
+                self.force_reboot = True
+                
+            raise
```

Comment:
Have you validated that the exception path out of here ensures that a reboot actually happens? (vs. terminating the execution completely with a failure)

Reply:
It looks like it is terminating the execution completely with a failure. I'll look into making sure that it actually reboots.

Reply:
Adjusted so that the error is only thrown if the error persists after a reboot; otherwise, throwing the error before a reboot will prevent the machine from rebooting.